### DISCUSSION TITLE
Images in Tooltips

### DISCUSSION THREAD

**Discussion by Posvar** [microsoft/SandDance#668] [Q&A] [answered]
I have thousands of images with lots of metadata. Each image is super small (12x12 px, 200-500 bytes).
SandDance is great for visualizing the data (and using a CSS Color column, for example, to represent a key attribute common across all images). But I'd love to use this visualization as a launch-pad to also view the image associated with the "cube". I tried including the datauri of each image as a column in CSV to no avail:
<img width="232" alt="image" src="https://github.com/microsoft/SandDance/assets/37360346/8f32a98c-8f08-49de-aa48-295e7712ee1a">

Any workarounds here that anyone has found to connect this visualization to images?
- In 2d format - the ideal would be to allow for the 2d tiles to be filled by the image (datauri?) - (similar to deepzoom)
- In 3d format - the ideal would to display the image at top of tooltip.
- At very least - having the tooltip display a selectable hyperlink to an image located elsewhere on web.

**Comment by Posvar** [comment]
Also no luck visualizing the PNG by inserting HTML into a data field:
<img width="208" alt="image" src="https://github.com/microsoft/SandDance/assets/37360346/8162616b-a73f-4f69-b3a1-e512cf85a2f4">

**Reply by danmarshall** [reply, marked as answer]
If you're using SandDance as a component in your own app, you can [provide your own tooltip implementation](https://github.com/microsoft/SandDance/issues/418#issuecomment-1191717495). If not, may I ask what context you're using it (online web app / VSCode extension / Azure Data Studio / other?)

I wonder how we can generalize this for other functionality - for example hyperlinks, or perhaps video.

**Reply by Posvar** [reply]
This was helpful to eventually figure it out!
